src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -713,6 +716,17 @@ struct
             (Error.to_string_hum e) ) ;
       Deferred.return (to_staged_ledger_or_error r)
     in
+    let () =
+      try
+        Coda_metrics.(
+          Gauge.set Snark_work.scan_state_snark_jobs
+            (Float.of_int
+               (List.length (Scan_state.all_work_pairs_exn scan_state'))))
```

Comment:
Is there any context where this *should* throw an exception? If it's always a bug we shouldn't catch the exception, the daemon should crash.

Reply:
This will throw an exception in the rpc call when snark works are requested. But we don't need to do that here when adding metrics?

Reply:
I don't understand. `all_work_pairs_exn` doesn't do an RPC, we're just querying a local data structure.

Reply:
It doesn't here. `all_work_pairs_exn` is used to fetch all the jobs for when snark workers request them over RPC